tests/array_api/util_functions.py

```diff
@@ -69,12 +70,16 @@ def test_clip_errors(self):
         ):
             xp.clip(a, 10, 90)
 
-    @pytest.mark.parametrize("dtype", DTYPES)
+    @pytest.mark.parametrize("dtype", DTYPES_WITH_BOOL)
```

Comment:
Continuing from the other PR, maybe you are just adding bool dtype to everything gradually :)

Reply:
Yes, `diff` specifically states in it's [documentation](https://numpy.org/doc/1.26/reference/generated/numpy.diff.html#numpy.diff) what the behavior should be for `bool` arrays. So I added it to the test to make sure we have that same behavior as `numpy`.

Reply:
Yep. Thanks for covering that case! (Likewise in the other PR)

Reply:
Well it turns out I didn't actually do this correctly. I missed the check that disallows bool. I've fixed that and also removed the test case which errors out when a bool array is passed in.